cmd/npmExecuteTests.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/orchestrator"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+func npmExecuteTests(config npmExecuteTestsOptions, _ *telemetry.CustomData) {
+	c := command.Command{}
+
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+	err := runNpmExecuteTests(&config, &c)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Step execution failed")
+	}
+}
+
+func runNpmExecuteTests(config *npmExecuteTestsOptions, c command.ExecRunner) error {
+	type AppURL struct {
+		URL      string `json:"url"`
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}
+
+	appURLs := make(map[string]AppURL)
+	urlsRaw, ok := config.AppSecrets["urls"].([]interface{})
+	if ok {
+		for _, urlRaw := range urlsRaw {
+			urlMap := urlRaw.(map[string]interface{})
+			url := urlMap["url"].(string)
+			appURLs[url] = AppURL{
+				URL:      url,
+				Username: urlMap["username"].(string),
+				Password: urlMap["password"].(string),
+			}
+		}
+	}
```

Comment:
We can handle the single url case with the collection as well.

```suggestion
	if ok {
		for _, urlRaw := range urlsRaw {
			urlMap := urlRaw.(map[string]interface{})
			url := urlMap["url"].(string)
			appURLs[url] = AppURL{
				URL:      url,
				Username: urlMap["username"].(string),
				Password: urlMap["password"].(string),
			}
		}
	} else {
	appURLs[url] = AppURL{
 				URL:      config.BaseURL,
 			}
	}
```

Reply:
I think this complicates it a bit